packages/headless/src/api/knowledge/stream-answer-api.ts

```diff
@@ -226,6 +227,11 @@ export const answerApi = answerSlice.injectEndpoints({
             onerror: (error) => {
               throw error;
             },
+            onclose: () => {
+              updateCachedData((draft) => {
+                dispatch(setHasNoAnswer(!draft.generated));
+              });
+            },
```

Comment:
Since you did use the `endOfStream` pattern on the searchApi side. I propose to move the dispatch logic to the endOfstream equivalent here. https://github.com/coveo/ui-kit/pull/4961/files#diff-27f1c99f828e98c6791d0300bae32a5c8db83255573eb724c341ba96306dbe0bR161 
cc. @lbergeron

Reply:
@dmgauthier tif we move it, the dispatch will be called also when we have no answer generated, in our current logic the endofstream is called only when we have an answer generated

Reply:
>  if we move it, the dispatch will be called also when we have no answer generated,

Thats what I though we wanted to achieve. Dispatching an action when the model cannot answer after the stream event closes. 

All in all its no big deal, I just feel we'd best bundle all the "stream has ended" dispatches at one place since we already have one in the endOfStream event handler.

Reply:
If you cannot move it without heavily modifiyng your current working logic, NVM it :ok_hand: